pkg/vendir/fetch/hg/hg.go

```diff
@@ -35,27 +51,60 @@ type HgInfo struct {
 	ChangeSetTitle string
 }
 
-func (t *Hg) Retrieve(dstPath string, tempArea ctlfetch.TempArea) (HgInfo, error) {
-	if len(t.opts.URL) == 0 {
-		return HgInfo{}, fmt.Errorf("Expected non-empty URL")
+// CloneHasTargetRef returns true if the given clone contains the target
+// ref, and this ref is a revision id (not a tag or a branch)
+func (t *Hg) CloneHasTargetRef(dstPath string) bool {
```

Comment:
In cachID() we are saying we are not including ref but here we are matching content with revisionID. 

May be I am not able to understand it correctly.  @cdevienne can you please help me to understand it bit more?
Thanks

Reply:
When we have an entry in the cache, we know it is the right repository.
From there we need to make sure it already contains the right changeset, which can be done by simply fetching changes. But if we tracked by changeset ID, we can test for its presence directly, avoiding a network roundtrip. But we cannot do it for tags and branches as they may change each time we fetch changes.

Reply:
> But if we tracked by changeset ID, we can test for its presence directly, avoiding a network roundtrip

Are you saying that changesetID( commit ID in term of git) we are storing in cache to compare it later when we found the repo in cache?
If yes, then I found the contradictory statements when we are generating cacheID and when we are reading/matching details in CloneHasTargetRef()

I think once you will add e2e test might be more helpful to understand.